### PR TITLE
Use javax.naming.ldap.LdapName to escape DNs with JNDI

### PR DESCRIPTION
**Pull request by HarlemSquirrel**
I noticed an issue saving entries that have forward slashes in their DN.

```
Error: test_set_dn_with_unnormalized_dn_attribute_with_forward_slash(TestBase): ActiveLdap::EntryNotFound: Couldn't find User: DN: temp/user1: filter: ["uid", "temp/user1"]
```

I found this [SO post](https://stackoverflow.com/questions/11690529/forward-slashes-in-the-names-returned-by-jndi-query-to-ldap-server) which pointed my to `javax.naming.ldap.LdapName`

> The LDAP name parser in the JDK is conservative with respect to quoting rules, but it nevertheless produces "correct" names. Also, remember that the names of entries returned by NamingEnumerations are composite names that can be passed back to the Context and DirContext methods. So, if the name contains a character that conflicts with the composite name syntax (such as the forward slash character "/"), then the LDAP provider will provide an encoding to ensure that the slash character will be treated as part of the LDAP name rather than as a composite name separator.
- https://docs.oracle.com/javase/tutorial/jndi/ldap/faq.html#20

### CODE REVIEW

File: lib/active_ldap/adapter/jndi_connection.rb [hunk 1]
```diff
@@ -134,10 +134,8 @@ def search(options)
         else
           @context.set_request_controls([])
         end
-
-        escaped_base = escape_dn(base)
         loop do
-          @context.search(escaped_base, filter, controls).each do |search_result|
+          @context.search(escape_dn(base), filter, controls).each do |search_result|
```
Comment (kou) [inline lib/active_ldap/adapter/jndi_connection.rb:138]: Could you revert this change?
This may cause multiple `escape_dn` calls.

Comment (HarlemSquirrel) [reply lib/active_ldap/adapter/jndi_connection.rb]: Updated

**Comment by kou** [comment]
Thanks!

**Comment by HarlemSquirrel** [comment]
@kou Thank you! Do you have a moment for a new version release?

**Comment by kou** [comment]
Done.
